### ISSUE TITLE
Qemu got stuck because of RMP fault when I enable svsm.bin. Could you take a look?

### ISSUE THREAD

**Issue by Unik-lif** [AMDESE/linux-svsm#58]
Hello!
I was going through the whole process for installing linux-svsm recently on a server with AMD's SEV-SNP enabled. It builds well and I successfully get the directory named "snp-release-2023-08-12" and the corresponding 'tar.gz' file. I ran the following command and the qemu can launch the Guest VM successfully.
```
[host machine]: $ uname -a
Linux super-server 6.1.0-rc4-snp-host-4c33a31c6e15 #1 SMP PREEMPT_DYNAMIC Sat Aug 12 20:27:41 CST 2023 x86_64 x86_64 x86_64 GNU/Linux
[host machine]: $ ./launch-qemu.sh -hda guest.qcow2
[guest machine]: $ uname -a
Linux link-server 6.2.0-snp-guest-20e69de4dd39 #2 SMP PREEMPT_DYNAMIC Sat Aug 12 20:07:46 CST 2023 x86_64 x86_64 x86_64 GNU/Linux
```
However, as I wanted to launch with svsm.bin, the qemu simply got stuck.......
```
link@super-server:~/linux-svsm/scripts$ sudo ./launch-qemu.sh -hda guest.qcow2 -sev-snp -svsm ../svsm.bin
./usr/local/bin/qemu-system-x86_64 -enable-kvm -cpu EPYC-v4,host-phys-bits=true -smp 4 -machine type=q35,confidential-guest-support=sev0,memory-backend=ram1,kvm-type=protected,vmport=off,svsm=../svsm.bin -object memory-backend-memfd-private,id=ram1,size=4G,share=true -object sev-snp-guest,id=sev0,policy=0x30000,cbitpos=51,reduced-phys-bits=1,init-flags=0x4,host-data=b2l3bmNvd3FuY21wbXA -drive if=pflash,format=raw,unit=0,file=/home/link/linux-svsm/scripts/usr/local/share/qemu/OVMF_CODE.fd,readonly=on -drive if=pflash,format=raw,unit=1,file=./guest.fd -drive file=guest.qcow2,if=none,id=disk0,format=qcow2 -device virtio-scsi-pci,id=scsi0,disable-legacy=on,iommu_platform=true -device scsi-hd,drive=disk0 -nographic -monitor pty -monitor unix:monitor,server,nowait -netdev user,id=vmnic -device e1000,netdev=vmnic,romfile=
Mapping CTRL-C to CTRL-]
Launching VM ...
  /tmp/cmdline.69063
char device redirected to /dev/pts/2 (label compat_monitor0)
qemu-system-x86_64: warning: Setting vm_type: protected
qemu-system-x86_64: warning: Setting vm_type: protected
qemu-system-x86_64: warning: UPM mode enabled, discard=(null)
qemu-system-x86_64: warning: creating ROM device with private memory.
qemu-system-x86_64: warning: creating ROM device with private memory.
```
It stuck like this, nothing else happened. I know the message shown here is not helpful but qemu just simply stuck like this.

As I delete "-svsm ../svsm.bin", only with `-sev-snp` enabled, which means I ran the following command:
```
 ./launch-qemu.sh -hda guest.qcow2 -sev-snp
```
It also got stuck like above.

And here is the system log in `/var/log/syslog` as Qemu got stuck.
```
Aug 13 22:44:35 super-server x11vnc[5584]:  --- x11vnc loop: sleeping 2000 ms ---
Aug 13 22:44:37 super-server kernel: [ 9829.028170] kvm_amd: Unexpected RMP fault for GPA 0x800013c000, error_code 0x680000007
Aug 13 22:44:37 super-server kernel: [ 9834.031966] handle_rmp_page_fault: 2884137 callbacks suppressed
Aug 13 22:44:37 super-server kernel: [ 9834.031971] kvm_amd: Unexpected RMP fault for GPA 0x800013c000, error_code 0x680000007
Aug 13 22:44:37 super-server kernel: [ 9834.031977] kvm_amd: Unexpected RMP fault for GPA 0x800013c000, error_code 0x680000007
Aug 13 22:44:37 super-server kernel: [ 9834.031979] kvm_amd: Unexpected RMP fault for GPA 0x800013c000, error_code 0x680000007
Aug 13 22:44:37 super-server kernel: [ 9834.031982] kvm_amd: Unexpected RMP fault for GPA 0x800013c000, error_code 0x680000007
Aug 13 22:44:37 super-server kernel: [ 9834.031985] kvm_amd: Unexpected RMP fault for GPA 0x800013c000, error_code 0x680000007
Aug 13 22:44:37 super-server kernel: [ 9834.031987] kvm_amd: Unexpected RMP fault for GPA 0x800013c000, error_code 0x680000007
Aug 13 22:44:37 super-server kernel: [ 9834.031990] kvm_amd: Unexpected RMP fault for GPA 0x800013c000, error_code 0x680000007
Aug 13 22:44:37 super-server kernel: [ 9834.031993] kvm_amd: Unexpected RMP fault for GPA 0x800013c000, error_code 0x680000007
Aug 13 22:44:37 super-server kernel: [ 9834.031995] kvm_amd: Unexpected RMP fault for GPA 0x800013c000, error_code 0x680000007
```
I suspect that the error may have been caused by my previous actions of running 
```
`sudo apt-get update` and `sudo apt-get upgrade`
``` 
on the server. But According to the syslog, it is too strange to blame on `update && upgrade`. In fact I tried to roll back to former version of my software, but it doesn't help. 

Besides, as I finished `./install.sh` and reboot, my host OS can't make simple kernel module like helloworld, it says:
```
ERROR: Kernel configuration is invalid etc...
```
I wonder if you have any suggestions on how to resolve this issue....Or maybe it's time to repair our server...?..

Earlier this year, approximately on March 16, 2023, my colleague was able to successfully launch virtual machines on the same server, running the same version of QEMU and the operating system, with `-svsm` enabled. 

PS: after my 'update & upgrade', my colleague can't launch his project with svsm.bin either...
PS: I've tested the Qemu version of cocosvsm, our server can build the Host OS & OVMF & Qemu & Guest OS of cocosvsm successfully, but the qemu also got stuck when I input svsm.bin of linux-svsm project.....

Looking forward to your reply And great thanks for this project.

**Comment by tlendacky** [comment]
I'm not sure what the problem could be. Did you also use the AMDSEV repo to build support for SNP? If so, the SVSM and SNP trees aren't completely in sync so I would suggest rebuilding everything again based on the instructions in the SVSM README.

**Comment by Unik-lif** [comment]
Thank u for ur reply!
I check my building environment for running `./build.sh --package` and read linux-svsm README.md more closely, and I find that my `./build.sh --package` is running on  6.1.0-rc4-snp-host-4c33a31c6e15 OS, I guess I'd better switch to 5.15.0-46-generic as you mentioned here to run the build scripts.
I will do soon, hope this will help.

**Comment by Unik-lif** [comment]
Ahh..
After making sure that BIOS is properly set, I tried Linux 5.15.0-46-generic to run linux-svsm build,sh scripts, but the fault of qemu still exists. Maybe there are some problems in our machine now.....

**Comment by Unik-lif** [comment]
I fixed this problem by reinstall ubuntu 22.04.3 again and clear some files in `/`, maybe some lib trigger this problem.
